tests/proxy_lifetime_tests.cpp

```diff
@@ -641,10 +641,15 @@ TEST(ProxyLifetimeTests, TestMoveAssignment_FromValue_ToSelf) {
 #ifdef __clang__
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wself-move"
+#elif __GNUC__ && __GNUC__ >= 13
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wself-move"
 #endif  // __clang__
     p = std::move(p);
 #ifdef __clang__
 #pragma clang diagnostic pop
+#elif __GNUC__ && __GNUC__ >= 13
```

Comment:
1. Please use `defined(__GNUC__)` or `ifdef` rather than relying on implicit conversion to bool.
2. If adding the parameter does not have side effects to earlier versions of GCC, suggest removing `__GNUC__ >= 13` for simplicity.
3. Nit: Since the lines above looks similar to the new block, can we merge them to make the block shorter?

Reply:
changed to `defined(__GNUC__)`